### PR TITLE
[SCIP_PaPILO,SCIP] try papilo on more platforms, threadlocal soplex

### PR DESCRIPTION
**Pull request by matbesancon**

### CODE REVIEW

**Comment by matbesancon** [comment]
and trying to tackle https://github.com/JuliaPackaging/Yggdrasil/pull/6054

**Comment by giordano** [comment]
What exactly are you trying to do?  https://github.com/JuliaPackaging/Yggdrasil/pull/7321#issuecomment-1715826494

**Comment by matbesancon** [comment]
ah indeed :facepalm:

**Comment by matbesancon** [comment]
```
[16:49:20] /opt/bin/x86_64-w64-mingw32-libgfortran5-cxx03/x86_64-w64-mingw32-g++ --sysroot=/opt/x86_64-w64-mingw32/x86_64-w64-mingw32/sys-root/  -Wno-psabi -O3 -DNDEBUG -Wl,--whole-archive CMakeFiles/papilo-executable.dir/objects.a -Wl,--no-whole-archive -o ../../bin/papilo.exe -Wl,--out-implib,libpapilo.dll.a -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles/papilo-executable.dir/linklibs.rsp
[16:49:25] /opt/x86_64-w64-mingw32/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld: ../libclusol.a(lusol6b.f.obj):lusol6b.f:(.text+0x499): undefined reference to `_gfortran_st_write'
[16:49:25] /opt/x86_64-w64-mingw32/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld: ../libclusol.a(lusol6b.f.obj):lusol6b.f:(.text+0x4af): undefined reference to `_gfortran_transfer_integer_write'
```

current error seems to be linked to fortran somehow

**Comment by giordano** [comment]
Linking fortran code with g++ isn't a great idea.  Either link it with gfortran, or add `-lgfortran` to linker invocation

File: S/SCIP_PaPILO/build_tarballs.jl [hunk 1]
```diff
@@ -14,6 +14,8 @@ sources = [
 script = raw"""
 cd scipoptsuite*
 
+LDFLAGS="${LDFLAGS} -lgfortran"
```
Comment (giordano) [inline S/SCIP_PaPILO/build_tarballs.jl:17]: You have to export variables to have any effect on subprocesses
```suggestion
export LDFLAGS="-lgfortran"
```

File: S/SCIP/build_tarballs.jl [hunk 2]
```diff
@@ -22,6 +22,9 @@ elif [[ "${target}" == *-mingw* ]]; then
    export LDFLAGS=-L${libdir}
 fi
 
+# for soplex threadlocal
+CXXFLAGS="-DTHREADLOCAL=''"
```
Comment (matbesancon) [inline S/SCIP/build_tarballs.jl:26]:
```suggestion
export CXXFLAGS="-DTHREADLOCAL=''"
```

**Comment by matbesancon** [comment]
still no luck so far

**Comment by matbesancon** [comment]
@giordano if you have a clue about what's happening I haven't managed to find the root cause for the fortran linking error

File: S/SCIP_PaPILO/build_tarballs.jl [hunk 3]
```diff
@@ -14,6 +14,8 @@ sources = [
 script = raw"""
 cd scipoptsuite*
 
+export LDFLAGS="-lgfortran"
```
[the remark this answers is not in the document]
Comment (matbesancon) [reply S/SCIP_PaPILO/build_tarballs.jl:17]: cc @alexhoen

**Comment by matbesancon** [comment]
thanks @benlorenz 
Hoping that we finally nail down the windows bug now :/

**Comment by matbesancon** [comment]
should be good to go now

**Review by fingolfin — APPROVED** [review]

**Comment by matbesancon** [comment]
@fingolfin do you know why the SCIP_jll packaging was not triggered?
https://github.com/JuliaBinaryWrappers/SCIP_jll.jl/
this is still the previous version that's out instead of 800.0.401

**Comment by giordano** [comment]
The answer is always in the build log on master: https://buildkite.com/julialang/yggdrasil/builds/5798

**Comment by matbesancon** [comment]
Indeed, got bitten by it. Investigating in https://github.com/JuliaPackaging/Yggdrasil/pull/7501

**Comment by fingolfin** [comment]
Huh, it did pass all CI tests here before I merged it, though... did something changed in the toolchain between that CI run and the merge? Or why else did it start failing like this on PowerPC only:
```
[22:58:55] cd /workspace/srcdir/scipoptsuite-8.0.4/build/scip/src && /usr/bin/cmake -E cmake_symlink_library ../../lib/libscip.so.8.0.4.0 ../../lib/libscip.so.8.0 ../../lib/libscip.so
  | [22:58:55] /opt/powerpc64le-linux-gnu/bin/../lib/gcc/powerpc64le-linux-gnu/6.1.0/../../../../powerpc64le-linux-gnu/bin/ld: warning: libdl.so.2, needed by /workspace/destdir/lib/libipopt.so, not found (try using -rpath or -rpath-link)
  | [22:58:55] /workspace/destdir/lib/libblastrampoline.so.5: undefined reference to `dladdr@GLIBC_2.17'
  | [22:58:55] /workspace/destdir/lib/libipopt.so: undefined reference to `dlerror@GLIBC_2.17'
  | [22:58:55] /workspace/destdir/lib/libipopt.so: undefined reference to `dlclose@GLIBC_2.17'
  | [22:58:55] /workspace/destdir/lib/libipopt.so: undefined reference to `dlopen@GLIBC_2.17'
  | [22:58:55] /workspace/destdir/lib/libipopt.so: undefined reference to `dlsym@GLIBC_2.17'
```

**Comment by giordano** [comment]
There have been some changes in the toolchain in BinaryBuilderBase.jl in the last few days, but I can't easily think of something that'd affect this.

**Pull request merged**